components/lib/ripple/Ripple.js

```diff
@@ -77,13 +78,20 @@ export const Ripple = React.memo(
             getTarget: () => targetRef.current
         }));
 
-        useMountEffect(() => {
-            if (inkRef.current) {
+        useUpdateEffect(() => {
```

Comment:
Should this just be `React.useEffect` and not update effect so it happens no matter what?  Right now its only happening on Update AFTER the component is mounted does this work in React.Strict and non-Strict mode?

Reply:
I used to think that if context.ripple changes, Ripple will re-render 
But I just tested it and it won’t render [demo](https://codesandbox.io/p/sandbox/primereact-demo-forked-66cj8z?file=%2Fsrc%2Findex.js%3A14%2C17)
If this is the design, I think it can simplify like
```tsx
useEffect(()=>{
   setMounted(true);
},[])
 useUpdateEffect(() => {
            if ( isMounted && inkRef.current) {
                targetRef.current = getTarget();
                setDimensions();
                bindEvents();
            }
      
 }, [ isMounted]);
```

Reply:
I haven't tested the production environment yet next12 13 14 and  vite

Reply:
OK even simpler.

```js
useMountEffect(()=>{
   setMounted(true);
})
```

as that is exactly what useMountEffect does.